File: backend/app/schemas/video.py
from pydantic import BaseModel
from enum import Enum

class VideoStatus(str, Enum):
    draft = "draft"
    rendering = "rendering"
    complete = "complete"
    failed = "failed"

class NewVideo(BaseModel):
    title: str
    topic: str
    description: str
    assetId: str

class Video(BaseModel):
    id: str
    title: str
    topic: str
    description: str
    status: VideoStatus
    assetId: str
    progress: int
    script: str
    output_file: str

File: frontend/src/types/video.ts
export enum VideoStatus {
  Draft = "draft",
  Rendering = "rendering",
  Complete = "complete",
}

export interface GameplayAsset {
  title: string;
  assetId: string;
  assetPath: string;
}

export interface VideoItem {
  id: string;
  title: string;
  topic: string;
  description: string;
  status: VideoStatus;
  assetId: string;
  progress: number;
  script: string;
  outputFile: string;
}

export interface DashboardMetric {
  label: string;
  value: string;
  change: string;
}

export interface NewVideoFormValues {
  title: string;
  topic: string;
  assetId: string;
  description: string;
}
